.github/workflows/platformDeploy.yml

```diff
@@ -0,0 +1,270 @@
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
+
+      - name: Setup python
+        run: sudo apt-get install python3-setuptools
+
+      - name: Setup Cloudflare CLI
+        run: |
+          # Pip 21 doesn't support python 3.5, so use the version before it
+          sudo python3 -m pip install --upgrade pip==20.3.4
+          pip3 install wheel # need wheel before cloudflare, this is the only way to ensure order.
+          pip3 install cloudflare
+
+      - name: Configure AWS Credentials
+        # Version: 1.5.5
+        uses: aws-actions/configure-aws-credentials@e97d7fbc8e0e5af69631c13daa0f4b5a8d88165b
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Install node packages
+        uses: nick-invision/retry@7c68161adf97a48beb850a595b8784ec57a98cbb
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: npm ci
+
+      - name: Build web for production
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: npm run build
+
+      - name: Deploy production to S3
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: aws s3 cp --recursive --acl public-read $GITHUB_WORKSPACE/dist s3://expensify-cash/
+
+      - name: Build web for staging
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: npm run build-staging
+
+      - name: Deploy staging to S3
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: aws s3 cp --recursive --acl public-read $GITHUB_WORKSPACE/dist s3://staging-expensify-cash/
+
+      - name: Purge production Cloudflare cache
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'true' }}
+        run: /home/runner/.local/bin/cli4 --delete hosts=["expensify.cash"] /zones/:357b42c0989e3bed82f6d6400e8b9d66/purge_cache
+        env:
+          CF_API_KEY: ${{ secrets.CLOUDFLARE_TOKEN }}
+
+      - name: Purge staging Cloudflare cache
+        if: ${{ env.SHOULD_DEPLOY_PRODUCTION == 'false' }}
+        run: /home/runner/.local/bin/cli4 --delete hosts=["staging.expensify.cash"] /zones/:357b42c0989e3bed82f6d6400e8b9d66/purge_cache
+        env:
+          CF_API_KEY: ${{ secrets.CLOUDFLARE_TOKEN }}
+
+  postSlackMessage:
+    name: Post a Slack message when any platform fails to build or deploy
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    needs: [android, desktop, iOS, web]
+    steps:
+      - uses: 8398a7/action-slack@v3
+        name: Job failed Slack notification
+        with:
+          status: custom
+          fields: workflow, repo
+          custom_payload: |
+            {
+              channel: '#announce',
+              attachments: [{
+                color: "#DB4545",
+                pretext: `<!here>`,
+                text: `💥 ${process.env.AS_REPO} failed on ${process.env.AS_WORKFLOW} workflow 💥`,
```

Comment:
Maybe the message should list which jobs failed.

Reply:
Any ideas on how? The text is kinda hard to change as you can only use their environment variables I think https://action-slack.netlify.app/fields

Reply:
Meh, probably not worth the ugly bash it would take to make this work.

Reply:
I think something like this might work tho:

```bash
run: |
  FAILED_PLATFORMS=()
  ${{ needs.android.status === 'failed' }} && FAILED_PLATFORMS+="android"
  ${{ needs.ios.status === 'failed' }} && FAILED_PLATFORMS+="ios"
  ${{ needs.web.status === 'failed' }} && FAILED_PLATFORMS+="web"
  ${{ needs.desktop.status === 'failed' }} && FAILED_PLATFORMS+="desktop"
  echo "FAILED_PLATFORMS=$($FAILED_PLATFORMS | sed  's/ /, /g')" >> $GITHUB_ENV
```

Reply:
And then can you access `process.env.FAILED_PLATFORMS`?